### PR TITLE
Bug 1132682 - Convert Mozmill test 'remote/testSecurity/testMixedContentPage.js' to Marionette r=jmaher

### PR DESCRIPTION
**Pull request by Ethcelon**
This PR converts the following test into Marionette.
http://hg.mozilla.org/qa/mozmill-tests/file/default/firefox/tests/remote/testSecurity/testMixedContentPage.js
bugzilla: https://bugzilla.mozilla.org/show_bug.cgi?id=1132682

### CODE REVIEW

File: firefox_ui_tests/remote/security/test_mixed_content.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+
+class TestMixedContent(FirefoxTestCase):
+    def setUp(self):
+        FirefoxTestCase.setUp(self)
+
+        self.test_url = 'https://mozqa.com/data/firefox/security/mixedcontent.html'
+
+    def tearDown(self):
+        FirefoxTestCase.tearDown(self)
```
Comment (chmanchester) [inline firefox_ui_tests/remote/security/test_mixed_content.py]: You can leave out the tearDown method if it's empty.

File: firefox_ui_tests/remote/security/test_mixed_content.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+    def test_mixed_content(self):
+        with self.marionette.using_context('content'):
+            self.marionette.navigate(self.test_url)
+        favicon = self.browser.navbar.locationbar.favicon
+
+        def get_list_style_image(favicon):
```
Comment (chmanchester) [inline firefox_ui_tests/remote/security/test_mixed_content.py]: This is only called once, I don't think we need a separate inner function.

Comment (Ethcelon) [inline firefox_ui_tests/remote/security/test_mixed_content.py]: I'll remove the function. Thanks for the advice!

File: firefox_ui_tests/remote/security/test_mixed_content.py (hunk 1 above)
Comment (Ethcelon) [inline firefox_ui_tests/remote/security/test_mixed_content.py]: okay, I will remove it

Comment (Ethcelon) [inline firefox_ui_tests/remote/security/test_mixed_content.py]: I'll take it out

File: firefox_ui_tests/remote/security/test_mixed_content.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+            favicon_image = self.marionette.execute_script("""
+                    return arguments[0].ownerDocument.defaultView
+                    .getComputedStyle(arguments[0])
+                    .getPropertyValue('list-style-image')
+            """, script_args=[favicon])
```
Comment (chmanchester) [inline firefox_ui_tests/remote/security/test_mixed_content.py]: Our indentation convention would make this:

```python
favicon_image = self.marionette.execute_script("""
  return arguments[0].ownerDocument.defaultView
                     .getComputedStyle(arguments[0])
                     .getPropertyValue('list-style-image');
""", script_args=[favicon])
```

File: firefox_ui_tests/remote/security/test_mixed_content.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    def test_mixed_content(self):
+        with self.marionette.using_context('content'):
+            self.marionette.navigate(self.test_url)
+        favicon = self.browser.navbar.locationbar.favicon
+
+        def get_list_style_image(favicon):
+            favicon_image = self.marionette.execute_script("""
+                    return arguments[0].ownerDocument.defaultView
+                    .getComputedStyle(arguments[0])
+                    .getPropertyValue('list-style-image')
```
Comment (chmanchester) [inline firefox_ui_tests/remote/security/test_mixed_content.py]: Please add a semicolon at the end of this line.

File: firefox_ui_tests/remote/security/test_mixed_content.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        favicon_image = get_list_style_image(favicon)
+        try:
+            is_mixed = favicon_image.index(
+                "identity-icons-https-mixed-display")
+            if is_mixed:
+                found = True
+        except Exception:
+            found = False
+        finally:
+            self.assertIs(found, True)
```
Comment (chmanchester) [inline firefox_ui_tests/remote/security/test_mixed_content.py]: I think the try...except...finally sequence be replaced with 
```python
self.assertIn("identity-icons-https-mixed-display", favicon_image)
```
The unittest library has lots of helpers documented here: https://docs.python.org/2/library/unittest.html

File: firefox_ui_tests/remote/security/test_mixed_content.py [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        def identity_popup_displayed(self):
+            return identity_popup.popup.is_displayed()
+
+        self.wait_for_condition(identity_popup_displayed)
+        encryption_label = identity_popup.encryption_label
+        label_text = encryption_label.text
+        prop = self.browser._l10n.get_property(
+            ['chrome://browser/locale/browser.properties'],
+            'identity.broken_loaded'
+            )
```
Comment (chmanchester) [inline firefox_ui_tests/remote/security/test_mixed_content.py]: Does this work as
```python
self.browser.get_property('identity.broken_loaded')
``` 
?

If not let me know, there might be a bug in our library.

File: firefox_ui_tests/remote/security/test_mixed_content.py (hunk 5 above)
Comment (Ethcelon) [inline firefox_ui_tests/remote/security/test_mixed_content.py]: Thanks for the information :) I have changed it! I wasn't exactly sure if it was used for strings or lists so I didn't use it.

File: firefox_ui_tests/remote/security/test_mixed_content.py (hunk 6 above)
Comment (Ethcelon) [inline firefox_ui_tests/remote/security/test_mixed_content.py]: @chmanchester No it does not. I tried that before using ._l10n.get_property

| MarionetteException: MarionetteException: DTD Entity not found: identity.broken_loaded

Comment (chmanchester) [inline firefox_ui_tests/remote/security/test_mixed_content.py]: Thanks for checking that. There is in fact a bug in our library I just filed as bug 1135248. Leave this for now and we'll update it when that bug is fixed.

File: firefox_ui_tests/remote/security/test_mixed_content.py (hunk 3 above)
Comment (chmanchester) [inline firefox_ui_tests/remote/security/test_mixed_content.py]: @Ethcelon did you see this comment? I think we had a mid-air collision, sorry about that.

**Comment by Ethcelon** [comment]
@chmanchester No it does not. I tried that before using ._l10n.get_property

| MarionetteException: MarionetteException: DTD Entity not found: identity.broken_loaded

File: firefox_ui_tests/remote/security/test_mixed_content.py (hunk 3 above)
Comment (Ethcelon) [inline firefox_ui_tests/remote/security/test_mixed_content.py]: yes i did, I changed the code to put all the dots neatly on a line

File: firefox_ui_tests/remote/security/test_mixed_content.py (hunk 4 above)
Comment (Ethcelon) [inline firefox_ui_tests/remote/security/test_mixed_content.py]: Added!

File: firefox_ui_tests/remote/security/test_mixed_content.py [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+            self.marionette.navigate(self.test_url)
+        favicon = self.browser.navbar.locationbar.favicon
+
+        favicon_image = self.marionette.execute_script("""
+            return arguments[0].ownerDocument.defaultView
+                               .getComputedStyle(arguments[0])
+                               .getPropertyValue('list-style-image');
+        """, script_args=[favicon])
+
+        self.assertIn("identity-icons-https-mixed-display", favicon_image)
```
Comment (chmanchester) [inline firefox_ui_tests/remote/security/test_mixed_content.py]: Looking at the mozmill test there's a wait for this condition. The page is mostly loaded by the time we return from self.marionette.navigate, but it's possible the effect on the favicon occurs after that, so to be safe let's add a wait here as well.

**Comment by Ethcelon** [comment]
@chmanchester the build failed because of:
https://travis-ci.org/mozilla/firefox-ui-tests/jobs/51580048#L803

**Comment by Ethcelon** [comment]
build failed because of:
https://travis-ci.org/mozilla/firefox-ui-tests/jobs/51580048#L803

**Comment by Ethcelon** [comment]
sorry about this, somewhere I went horribly wrong

File: firefox_ui_tests/remote/security/test_mixed_content.py (hunk 6 above)
Comment (chmanchester) [inline firefox_ui_tests/remote/security/test_mixed_content.py]: The fix landed, you can update this if you want to before landing.

**Comment by jmaher** [comment]
congratulations on converting your first test!

**Comment by Ethcelon** [comment]
Thank you for all the help! Im looking forward to converting more @jmaher
@chmanchester @whimboo @galgeek

**Pull request closed without merging**